src/gt4py/next/program_processors/runners/dace_fieldview/transformations/util.py

```diff
@@ -0,0 +1,168 @@
+    never contain `end` nor `begin`. In case `end` is never found the function
+    will return `None`.
+
+    If `reverse` is set to `True` the function will start exploring at `end` and
+    follows the outgoing edges, i.e. the meaning of `end` and `begin` are swapped.
+
+    Args:
+        graph: The graph to operate on.
+        begin: The start of the DFS.
+        end: The terminator node of the DFS.
+        reverse: Perform a backward DFS.
+
+    Notes:
+        - The returned set will also contain the nodes of path that starts at
+            `begin` and ends at a node that is not `end`.
+    """
+
+    def next_nodes(node: dace_nodes.Node) -> Iterable[dace_nodes.Node]:
+        if reverse:
+            return (edge.src for edge in graph.in_edges(node))
+        return (edge.dst for edge in graph.out_edges(node))
+
+    if reverse:
+        begin, end = end, begin
+
+    to_visit: list[dace_nodes.Node] = [begin]
+    seen: set[dace_nodes.Node] = set()
+    found_end: bool = False
+
+    while len(to_visit) > 0:
+        n: dace_nodes.Node = to_visit.pop()
+        if n == end:
+            found_end = True
+            continue
+        elif n in seen:
+            continue
+        seen.add(n)
+        to_visit.extend(next_nodes(n))
+
+    if not found_end:
+        return None
+
+    seen.discard(begin)
+    return seen
+
+
+def find_downstream_consumers(
+    state: dace.SDFGState,
+    begin: dace_nodes.Node | dace_graph.MultiConnectorEdge[dace.Memlet],
+    only_tasklets: bool = False,
+    reverse: bool = False,
+) -> set[tuple[dace_nodes.Node, dace_graph.MultiConnectorEdge[dace.Memlet]]]:
+    """Find all downstream connectors of `begin`.
+
+    A consumer, in for this function, is any node that is neither an entry nor
+    an exit node. The function returns a set of pairs, the first element is the
+    node that acts as consumer and the second is the edge that leads to it.
+    By setting `only_tasklets` the nodes the function finds are only Tasklets.
+
+    To find this set the function starts a search at `begin`, however, it is also
+    possible to pass an edge as `begin`.
+    If `reverse` is `True` the function essentially finds the producers that are
+    upstream.
+
+    Args:
+        state: The state in which to look for the consumers.
+        begin: The initial node that from which the search starts.
+        only_tasklets: Return only Tasklets.
+        reverse: Follow the reverse direction.
+    """
+    if isinstance(begin, dace_graph.MultiConnectorEdge):
+        to_visit: list[dace_graph.MultiConnectorEdge[dace.Memlet]] = [begin]
+    elif reverse:
+        to_visit = list(state.in_edges(begin))
+    else:
+        to_visit = list(state.out_edges(begin))
+    seen: set[dace_graph.MultiConnectorEdge[dace.Memlet]] = set()
+    found: set[tuple[dace_nodes.Node, dace_graph.MultiConnectorEdge[dace.Memlet]]] = set()
+
+    while len(to_visit) != 0:
```

Comment:
For coherence with the code above:
```suggestion
    while len(to_visit) > 0:
```